### PR TITLE
Add CI Builds to PRs and pushes to main

### PR DESCRIPTION
**Pull request by mikeage**
Build Android/Oculus, Win64/Monoscopic, Win64/SteamVR, Linux64/Monoscopic, Linux64/SteamVR, and OSX/Monoscopic (SteamVR is not support on Mac any more)

Included commits:
* Install Oculus v23.1 unitypackage
* Install TMP Essential Resources from v2.0.1
* Add and ignore (mandatory) option -buildVersion
* Add and ignore (mandatory) option -android*
* Rename Tiltasaurus.json to tiltasaurus.json to match the filename in
  both the .cs and .py files
* Accept -customBuildPath as an alternative to -btb-out
* Pass -btb-out explicitly (TODO; why can't we use -customBuildPath)
* Add code to replace InitializeOnLoad (not called in CI)
* Explicitly call OVRPluginUpdater prior to build (normally done in the editor)
* Add -setDefaultPlatformTextureFormat astc for Android builds
* Add -btb-github to change Display Name and App Identifier to contain
  'github' to not conflict with the official OpenBrush apk
* Add the github SHA to the build stamp

### CODE REVIEW

**Comment by mikeage** [comment]
This works, but only if we also have something like PR #18 (discussion in #23), as it compiles with the default packages of Unity 2019.4.18f, which is Android SDK version 29.

It also needs either a Github secret (safer, but will only run on commits / PRs from the main repo, not forks!) or an embedded license file (probably safe enough since it's from a free unity license anyway; it's not really a secret when they give it out!)

See https://game.ci/docs/github/getting-started for more information

File: Assets/Editor/BuildTiltBrush.cs [hunk 1]
```diff
@@ -651,6 +664,14 @@ class TempSetCommandLineOnlyPlayerSettings : IDisposable {
         } else if (args[i] == "-btb-increment-bundle-version") {
           // Don't restore this, because user might want to check in this change?
           PlayerSettings.Android.bundleVersionCode += 1;
+        } else if (args[i] == "-buildVersion") {
+          // TODO: do we want to do anything with this? Can we use it instead of the version string set externally?
+          i++;
+        } else if (args[i] == "-androidVersionCode" || args[i] == "-androidKeystoreName" || args[i] == "-androidKeystorePass" || args[i] == "-androidKeyaliasName" || args[i] == "-androidKeyaliasPass") {
```
Comment (TimAidley) [inline Assets/Editor/BuildTiltBrush.cs:670]: nit:
On Tilt Brush we limited all code lines to 100 characters in length. I don't think we've made any decision on whether to continue with that on Open Brush, but if we want to keep consistent we should do the same.

File: Assets/Editor/BuildTiltBrush.cs [hunk 2]
```diff
@@ -608,6 +610,13 @@ class TempSetCommandLineOnlyPlayerSettings : IDisposable {
     };
     string keystoreName = null;
     string keyaliasName = null;
+
+#if OCULUS_SUPPORTED
+    // Call these once to create the files. Normally (i.e., in a GUI build), they're created with [UnityEditor.InitializeOnLoad], but in case they're missing, like in CI, make sure they're there!
```
Comment (TimAidley) [inline Assets/Editor/BuildTiltBrush.cs:615]: nit: line length

**Review by TimAidley — APPROVED** [review]
See slight nit about line length;

otherwise; LGTM.

**Comment by mikeskydev** [comment]
Line length seems sensible for readability! Thanks for reviewing @TimAidley, will set as ready. @mikeage Feel free to DM me to walk through secrets/licence file steps.

File: Assets/Editor/BuildTiltBrush.cs (hunk 1 above)
Comment (mikeage) [reply Assets/Editor/BuildTiltBrush.cs:670]: Sure, makes sense. I’m going to open a separate PR for automated checking this as well. No reason for a reviewer to need to look for this type of thing!

**Comment by mikeage** [comment]
I wasn't sure if this would run or not yet, but it did. To get the activation file, go to https://github.com/mikeage/open-brush/actions/runs/634471317 (valid for 90 days) and download the artifact called "Unity_v2019.4.18f1.alf" and upload that to at license.unity3d.com (for a free license)

**Review by mikeskydev — APPROVED** [review]

**Pull request merged**